server/server.go

```diff
@@ -220,49 +221,53 @@ func (s *Server) handleGetMh(w lookupResponseWriter, r *http.Request) {
 	}
 	mh := w.Key()
 	log := logger.With("multihash", mh)
-	ctx := context.Background()
+	ctx, cancel := context.WithCancel(r.Context())
```

Comment:
This additional context may not be necessary since the context will get closed when the client connection is closed.  However, if the request context does not get closed right away, then this will allow the response reader to exit sooner. So, this may not be needed, but is here to make sure the context gets canceled ASAP.